src/primitive_core.cc

```diff
@@ -2225,13 +2225,11 @@ PRIMITIVE(literal_index) {
   ARGS(Object, o);
   auto null_object = process->program()->null_object();
   if (!o->is_heap_object()) return null_object;
-  auto heap_object = HeapObject::cast(o);
-  if (!Heap::in_read_only_program_heap(heap_object, process->object_heap())) return null_object;
```

Comment:
Was it necessary to fix this or did you just decide to remove the fast case? I assume the fast (fail) case isn't used by toitp.

Reply:
We have circular dependencies between .a files - not sure why we didn't have this issue before, but this makes it link.

Reply:
Maybe it wasn't necessary.